edward/inferences/inference.py

```diff
@@ -132,8 +132,15 @@ def __init__(self, latent_vars=None, data=None, model_wrapper=None):
               raise TypeError("Observed variable bindings do not have same "
                               "shape.")
 
-            # If value is a np.ndarray, store it in the graph.
-            ph = tf.placeholder(tf.float32, value.shape)
```

Comment:
The TensorFlow placeholder needs to have the same basic type (e.g. integer or float) as the numpy array for which it's serving as a placeholder.